src/nimble_mpi.cc

```diff
@@ -421,36 +421,32 @@ int ExplicitTimeIntegrator(
 
   nimble::ModelData &model_data = to_ModelData(data_manager.GetMacroScaleData());
 
-  int reference_coordinate_field_id = model_data.GetFieldId("reference_coordinate");
-  int displacement_field_id = model_data.GetFieldId("displacement");
-  int velocity_field_id =  model_data.GetFieldId("velocity");
-  int acceleration_field_id =  model_data.GetFieldId("acceleration");
-  int internal_force_field_id =  model_data.GetFieldId("internal_force");
-  int external_force_field_id =  model_data.GetFieldId("external_force");
-  int contact_force_field_id =  model_data.GetFieldId("contact_force");
-
   int status = 0;
+
   // Set up the global vectors
   unsigned int num_unknowns = num_nodes * mesh.GetDim();
-  double* reference_coordinate = model_data.GetNodeData(reference_coordinate_field_id);
-  double* displacement = model_data.GetNodeData(displacement_field_id);
-  double* velocity = model_data.GetNodeData(velocity_field_id);
-  double* acceleration = model_data.GetNodeData(acceleration_field_id);
-  double* internal_force = model_data.GetNodeData(internal_force_field_id);
-  double* external_force = model_data.GetNodeData(external_force_field_id);
-  double* contact_force = (contact_enabled) ? model_data.GetNodeData(contact_force_field_id)
-                           : nullptr;
+
+  auto reference_coordinate = model_data.GetVectorNodeData("reference_coordinate");
+  auto displacement = model_data.GetVectorNodeData("displacement");
+  auto velocity = model_data.GetVectorNodeData("velocity");
+  auto acceleration = model_data.GetVectorNodeData("acceleration");
+  auto internal_force = model_data.GetVectorNodeData("internal_force");
+  auto external_force = model_data.GetVectorNodeData("external_force");
+
+  nimble::Viewify<2> contact_force;
+  if (contact_enabled)
+    contact_force = model_data.GetVectorNodeData("contact_force");
 
 #ifdef NIMBLE_HAVE_UQ
   auto uq_model = *( data_manager.GetUqModel() );
-  std::vector<Viewify> bc_offnom_velocity_views(0);
+  std::vector< nimble::Viewify<2> > bc_offnom_velocity_views(0);
   if(uq_model.Initialized()) {
     uq_model.Setup();
 /// FOR CALL TO BCS ===================
     int num_samples = uq_model.GetNumSamples();
     for(int nuq=0; nuq<num_samples; nuq++){
        double * v = uq_model.Velocities()[nuq];
-       bc_offnom_velocity_views.push_back( Viewify(v,3) );
+       bc_offnom_velocity_views.push_back( nimble::Viewify<2>(v, {0, 3}, {3, 1}) );
```

Comment:
why are the arguments "{0, 3}, {3, 1}" necessary vs "3"?

Reply:
The first array (here {0, 3}) is for the length in each dimension .
Actually 0 should be replaced with a non-null value.

The second array (here {3, 1}) is for the strides in each dimension.

I will fix that. I would like to move the UQ sections but I have not found the right solution yet.

Reply:
@rjones-sandia 
I do not know the length of `double * v = uq_model.Velocities()[nuq]` (since I do not have `nimble_uq.h`).